mergify_cli/ci_issues/junit_upload.py

```diff
@@ -0,0 +1,32 @@
+from mergify_cli import console
+from mergify_cli.ci_issues import utils
+
+
+async def upload(  # noqa: PLR0913, PLR0917
+    api_url: str,
+    token: str,
+    repository: str,
+    head_sha: str,
+    job_name: str,
+    provider: str | None,
+    files: tuple[str, ...],
+) -> None:
+    files_to_upload = utils.get_files_to_upload(files)
+
+    form_data = {
+        "head_sha": head_sha,
+        "name": job_name,
+    }
+    if provider is not None:
+        form_data["provider"] = provider
+
+    async with utils.get_ci_issues_client(api_url, token) as client:
+        response = await client.post(
+            f"/repos/{repository}/ci_issues_upload",
+            data=form_data,
+            files=files_to_upload,
+        )
+
```

Comment:
handle status code != 200?

Reply:
I was thinking about letting them break to fail the script and get the stacktrace for feedback, response's detail is already printed, wdyt?